yt/frontends/gadget/io.py

```diff
@@ -77,13 +80,58 @@ def _yield_coordinates(self, data_file, needed_ptype=None):
             yield key, pos
         f.close()
 
+    def _generate_smoothing_length(self, data_files, kdtree):
+        if not self.ds.gen_hsmls:
+            return
+        # TODO: We need to do something like this here but I'm not sure how
```

Comment:
Could you elaborate what you are uncertain about here ?

Reply:
This was copied over directly from the Tipsy frontend. It is supposed to prevent the smoothing length files from being regenerated if they are already there and are consistent with the dataset. But in this case 1) the files are HDF5 and not unformatted like Tipsy and 2) in this case there are also going to be multiple HDF5 files for the smoothing length, being the same number as the number of files in the dataset. So we have to adjust the logic here but I am not sure how.